### PR TITLE
Add querying helpers

### PR DESCRIPTION
**Pull request by zoul**
These are helpers to query values stored in a `JSON` value proposed by @cjmconie in #2. Cameron, is this OK? I have mostly just copied your code and added some tests.

### CODE REVIEW

File: GenericJSON/Querying.swift [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    /// Return the array value if this is an `.array`, otherwise `nil`
+    public var arrayValue: [JSON]? {
+        if case .array(let value) = self {
+            return value
+        }
+        return nil
+    }
+
+    /// Return `true` iff this is `.null`
```
Comment (cjmconie) [inline GenericJSON/Querying.swift:45]: Typo "iff"

File: GenericJSON/Querying.swift [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            return true
+        }
+        return false
+    }
+
+    /// If this is an `.array`, return item at index
+    ///
+    /// If this is not an `.array` or the index is out of bounds, returns `nil`.
+    public subscript(index: Int) -> JSON? {
+        if case .array(let arr) = self, index < arr.count {
```
Comment (cjmconie) [inline GenericJSON/Querying.swift]: We check the upper bound of the array here, does it make sense to check the lower bound (i.e. index >= 0) too? Currently, an overflow in one direction returns `nil` and the other direction will trigger a fatal index out of range exception.

File: GenericJSON/Querying.swift (hunk 1 above)
Comment (zoul) [reply GenericJSON/Querying.swift:45]: http://en.wikipedia.org/wiki/If_and_only_if

File: GenericJSON/Querying.swift (hunk 2 above)
Comment (zoul) [reply GenericJSON/Querying.swift]: Good catch! Will do.

**Pull request merged**